comp/collector/collector/collectorimpl/internal/middleware/check_wrapper.go

```diff
@@ -32,20 +36,29 @@ type CheckWrapper struct {
 }
 
 // NewCheckWrapper returns a wrapped check.
-func NewCheckWrapper(inner check.Check, senderManager sender.SenderManager) *CheckWrapper {
+func NewCheckWrapper(inner check.Check, senderManager sender.SenderManager, agentTelemetry option.Option[agenttelemetry.Component]) *CheckWrapper {
 	return &CheckWrapper{
-		inner:         inner,
-		senderManager: senderManager,
+		inner:          inner,
+		senderManager:  senderManager,
+		agentTelemetry: agentTelemetry,
 	}
 }
 
 // Run implements Check#Run
-func (c *CheckWrapper) Run() error {
+func (c *CheckWrapper) Run() (err error) {
 	c.runM.Lock()
 	defer c.runM.Unlock()
 	if c.done {
 		return nil
 	}
+
+	// Start telemetry span if telemetry is enabled
+	if telemetry, isSet := c.agentTelemetry.Get(); isSet {
+		span, _ := telemetry.StartStartupSpan(fmt.Sprintf("check.%s", c.inner.String()))
+		defer span.Finish(err)
+	}
```

Comment:
In the first implementation, we only added to the Python checks, now we are adding to every check. Is that intended?

Reply:
Yes, we get more observability data there